extensions/amp-story/1.0/amp-story-page.js

```diff
@@ -1658,4 +1682,42 @@ export class AmpStoryPage extends AMP.BaseElement {
       );
     });
   }
+
+  /**
+   * Listens for changes on captions if there are tracks on videos and page is active.
+   * @private
+   */
+  initializeCaptionsListener_() {
+    this.hasLoadedVideoWith_((video) => video.querySelector('track')).then(
```

Comment:
(also, his helper doesn't even check if the video has loaded at all)

Reply:
It waits for the `buildCallback` to finish due to the `getImpl()` call for cached videos with audio, which is what `Loaded` was trying to mean in this context. I'll try to clean this logic up anyways, so it's more clear